ios/NumberTileGame/HighScoreArchiver.m

```diff
@@ -0,0 +1,68 @@
+//
+//  HighScoreArchiver.m
+//  NumberTileGame
+//
+//  Created by Michael Prouty on 10/30/19.
+//
+
+#import "HighScoreArchiver.h"
+static NSString *const highScoreArchivePath = @"highScoreArchive";
+
+@implementation HighScoreArchiver
+
++ (NSString*)filePath {
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+    NSString *documentsDirectoryPath = [paths objectAtIndex:0];
+    NSString *path = [documentsDirectoryPath stringByAppendingPathComponent:highScoreArchivePath];
+    // NSLog(@"File path: %@", path);
+    return path;
+}
+
++ (void)addScore:(NSNumber*)value {
+    NSLog(@"new score added of: %@", value);
+
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+    dateFormatter.dateStyle = NSDateFormatterShortStyle;
+    dateFormatter.timeStyle = NSDateFormatterShortStyle;
+    NSString *dateString = [dateFormatter stringFromDate:[NSDate date]];
+
+    NSDictionary *score = @{
+                                 @"value" : value,
+                                 @"createdAt" : dateString
```

Comment:
Why are we saving the timestamp as a string? NSDateFormatter is locale dependent, meaning that if the user changes their region settings the existing scores will remain in the old date format, and new scores will be saved with a new date format. The correct thing would be to save the date as a date, or as a number, or as a locale-independent string (see [NSIOS8601DateFormatter](https://developer.apple.com/documentation/foundation/nsiso8601dateformatter) and/or [this Tech Note](https://developer.apple.com/library/archive/qa/qa1480/_index.html)).

I know that NSDate supports the NSCoding protocol, so I’m guessing you did this because the date objects couldn't be easily passed back and forth between native iOS and JavaScript as easily as strings... is that correct?

Reply:
Alright, I switched this to pass an ISO8601 string.  Looks like the system timezone is used when parsing this on the RN thread (which makes sense).

> I’m guessing you did this because the date objects couldn't be easily passed back and forth between native iOS and JavaScript

Yes -- it looks like only standard JSON data types can be passed: string, number, boolean, array, object.  In this case, we can get away with just passing an ISO8601 string.  But for the app, we may need to come up with a JSON serialization strategy for our native objects (unless we can get around this by just having both threads talk to Realm...)

Further reading: https://medium.com/shoutem/ways-to-pass-objects-between-native-and-javascript-in-react-native-c3dcae7bf4f5